client/src/Components/HomePage/index.js

```diff
@@ -1,11 +1,41 @@
 import React, { Component } from "react"
 import Header from "../Header"
+import Search from "../Search"
+import axios from "axios"
 
 class HomePage extends Component {
+  state = {
+    pharmaciesResult: "",
+    medicineName: ""
+  }
+
+  changeInput = event => {
+    const { target } = event
+    const { value } = target
+    this.setState({ medicineName: value })
+  }
+
+  searchHandler = () => {
+    const { pharmaciesResult, medicineName } = this.state
+    // This should handle in backend
+    axios.get(`/api/medicine/${medicineName}`).then(({ data }) => {
+      this.setState({
+        pharmaciesResult: data
+      })
+    })
+    const { history } = this.props
+    history.push({ pathname: "/results", data: pharmaciesResult })
```

Comment:
you are passing an empty object here as you declaring the "pharmaicesResult" before you call the endpoint to update this value